super_calendar/models/super_calendar_configurator.py

```diff
@@ -0,0 +1,160 @@
+#    along with this program.  If not, see http://www.gnu.org/licenses
+#
+##############################################################################
+
+from openerp import models, fields, api
+from openerp import _
+from mako.template import Template
+from datetime import datetime
+from openerp import tools
+from openerp.tools.safe_eval import safe_eval
+import logging
+
+
+_logger = logging.getLogger(__name__)
+
+
+class SuperCalendarConfigurator(models.Model):
+    _name = 'super.calendar.configurator'
+
+    name = fields.Char(
+        string='Name',
+        required=True,
+    )
+    line_ids = fields.One2many(
+        comodel_name='super.calendar.configurator.line',
+        inverse_name='configurator_id',
+        string='Lines',
+    )
+
+    def _clear_super_calendar_records(self):
+        """
+        Remove old super_calendar records
+        """
+        super_calendar_pool = self.env['super.calendar']
+        super_calendar_list = super_calendar_pool.search([])
+        super_calendar_list.unlink()
+
+    @api.multi
+    def generate_calendar_records(self):
+        """
+        At every CRON execution, every 'super calendar' data is deleted and
+        regenerated again.
+        """
+
+        # Remove old records
+        self._clear_super_calendar_records()
+
+        # Rebuild all calendar records
+        configurator_list = self.search([])
+        for configurator in configurator_list:
+            for line in configurator.line_ids:
+                configurator._generate_record_from_line(line)
+        _logger.info('Calendar generated')
+        return True
+
+    @api.multi
+    def _generate_record_from_line(self, line):
+        """
+        Create super_calendar records from super_calendar_configurator_line
+        objects.
+        """
+        super_calendar_pool = self.env['super.calendar']
+        values = self._get_record_values_from_line(line)
+        for record in values:
+            super_calendar_pool.create(values[record])
+
+    @api.multi
+    def _get_record_values_from_line(self, line):
+        """
+        Get super_calendar fields values from super_calendar_configurator_line
+        objects.
+        Check if the User value is a res.users.
+        """
+        res = {}
+        current_pool = self.env[line.name.model]
+        domain = line.domain and safe_eval(line.domain) or []
+        current_record_list = current_pool.search(domain)
+        for cur_rec in current_record_list:
+            f_user = line.user_field_id.name
+            f_descr = line.description_field_id.name
+            f_date_start = line.date_start_field_id.name
+            f_date_stop = line.date_stop_field_id.name
+            f_duration = line.duration_field_id.name
+            if (f_user and
+                    cur_rec[f_user] and
+                    cur_rec[f_user]._model._name != 'res.users'):
+                raise Exception(
```

Comment:
Looks like the wrong type of error is being raised here. Try something from openerp.exceptions